### PR TITLE
1.2.0 Feature: Realtime: Connection Recovery

### PR DESCRIPTION
**Pull request by QuintinWillison**
There was no benefit in having a discrete feature node for the connection recovery string client option.
It's implicitly required for this functionality to work at all, and there's potential that client libraries could decide to offer a more idiomatic API that means it's not even framed via any client options API.

One of a family of pull requests addressing #3.

### CODE REVIEW

**Comment by QuintinWillison** [comment]
Via #25 I've [discovered](https://github.com/ably/features/runs/7832838300?check_suite_focus=true) what Prettier's grumbling at:

```
diff --git a/sdk.yaml b/sdk.yaml
index d5e1074..847b9c8 100644
--- a/sdk.yaml
+++ b/sdk.yaml
@@ -296,7 +296,8 @@ Realtime:
         Mechanisms to get connection state synchronously, as well as to receive events asynchronously when connection state changes.
         Also provides access to an error reason object in case of failure.
     Recovery:
-      .specification: [RTN9, RTN10, RTN[16](https://github.com/ably/features/runs/7832738512?check_suite_focus=true#step:8:17), RTC1c, TO3i, RTN11d, RTN15a, RTN15b, RTN15d]
+      .specification:
+        [RTN9, RTN10, RTN16, RTC1c, TO3i, RTN11d, RTN15a, RTN15b, RTN15d]
       .synopsis: |
         Capability to recover a connection by explicitly providing a recovery key when instantiating a Realtime client instance,
         where that recovery key has been sourced from a previous instance.
```

I still have no idea why this does not fail on my local machine. 🤷 

Regardless, I'm going to push up a refined variant of [my Prettier experiments PR](https://github.com/ably/features/pull/25) that can underly this PR in order to prove the fix...

**Comment by lmars** [comment]
I don't really know how connection recovery works, so I defer to others who do know to review this one.

**Comment by QuintinWillison** [comment]
@SimonWoolf I realise that you've perhaps got less context to review this holistically, in the context of the canonical features as a whole, but could you please review in respect of spec points referenced and the synopsis?

**Review by SimonWoolf — APPROVED** [review]
Not sure-familiar with the arrangement of features in the feature repo but seems plausible. One comment.

File: sdk.yaml [hunk 1]
```diff
@@ -290,6 +290,17 @@ Realtime:
...
+      .specification:
+        [RTN9, RTN10, RTN16, RTC1c, TO3i, RTN11d, RTN15a, RTN15b, RTN15d]
+      .synopsis: |
+        Capability to recover a connection by explicitly providing a recovery key when instantiating a Realtime client instance,
+        where that recovery key has been sourced from a previous instance.
+
+        Client instances provide:
+        - a unique private connection `key` provided by the Ably service that is used to reconnect and retain connection state following an unexpected disconnection
+        - a `serial` attribute, set from the `connectionSerial` received from the Ably service when `CONNECTED` state is confirmed
+        - a `recoveryKey`, commposed from the connection key, latest connection serial (`connectionSerial`) and the current message serial (`msgSerial`)
```
[the remark this answers is not in the document]
Comment (QuintinWillison) [reply sdk.yaml:306]: https://github.com/ably/features/pull/24/commits/f08401d7ff501f8df033a5cfab9aad17c5cb8b9a

I'm going to resolve this conversation on @SimonWoolf's behalf, in order to enable me to merge this pull request sooner.

**Pull request merged**